### DISCUSSION TITLE
Custom OAuth2 by "social login" app authentication with authentik

### DISCUSSION THREAD

**Discussion by Mathusoothanan-k** [nextcloud/all-in-one#5560] [Questions] [answered]
Hello everyone,

I'm currently integrating Authentik and Nextcloud using Custom OAuth2 in Nextcloud's social login app.To configure Authentik, I've created an application named 'Nextcloud' with the settings shown in the attached screenshot. When I access this application, it should redirect me to "https://nextcloud.example.com/apps/sociallogin/custom_oauth2/Authentik".
![authentik](https://github.com/user-attachments/assets/9d025057-0632-4aa8-9c04-6d091d66a6a9)

After that I created the provider for the 'Nextcloud' application in Authentik. 
![nextcloud provider](https://github.com/user-attachments/assets/d731731c-cf29-41ee-8acb-c99b28af362f)
![nextcloud provider 2](https://github.com/user-attachments/assets/ddd5f0e0-0c07-44f4-a9c2-c3a0d5fd42fa)
OAuth2 configuration is complete on the Authentik side. 

I am now proceeding to configure custom OAuth2 settings in Nextcloud.
![nextconfig](https://github.com/user-attachments/assets/42e54255-57a1-4557-be8c-02d474d41de6)

The issue arises when I click the Authentik 'Nextcloud' application. It should redirect to "https://nextcloud.example.com/apps/sociallogin/custom_oauth2/Authentik", but instead, it logs me directly into "nextcloud.example.com" with Authentik user credentials.

This "https://nextcloud.example.com/apps/sociallogin/custom_oauth2/Authentik" link bypasses the login UI, achieving Single Sign-On (SSO) with one click, similar to SAML."
![Screenshot 2024-11-09 at 13-34-14 Dashboard - Nextcloud](https://github.com/user-attachments/assets/64c36e16-c5c0-4145-b6e7-035f4f4c5c2b)

After closing my Nextcloud site ("http://nextcloud.example.com") without logging out, I revisited Authentik and clicked the 'Nextcloud' application again. However, it threw an error because I was still logged in to Nextcloud "http://nextcloud.example.com". For each reload the error could be vary.

![error 1](https://github.com/user-attachments/assets/65389939-bd82-4226-93f4-27b8534cf7dc)
![error 2](https://github.com/user-attachments/assets/f3c0aa3d-f026-45ac-bb8f-656fb2c72400)
![error 3](https://github.com/user-attachments/assets/46266944-a4e2-4a96-9ac9-12beca59bd16)

Interestingly, when I performed the same steps with Frappe (using social login and OAuth2), I was allowed to log in again without issues. But Nextcloud throws an error in this scenario.

**Comment by szaimen** [comment, marked as answer]
Hi, please post this at https://help.nextcloud.com. Thanks!

**Reply by Mathusoothanan-k** [reply]
yeah ok, I posted the same thing in https://help.nextcloud.com/t/custom-oauth2-by-social-login-app-authentication-with-authentik/209616
